http_service/app.py

```diff
@@ -3,37 +3,80 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
+import json
 import logging
 import os
+from collections import UserDict, defaultdict
+from multiprocessing.pool import AsyncResult, Pool
 
-from flask import Flask, current_app, jsonify, request
+from flask import Flask, jsonify, request
 
-from bugbug import bugzilla
-
-from .models import load_model
+from .models import classify_bug
 
 API_TOKEN = "X-Api-Key"
 
 application = Flask(__name__)
 
-bugzilla.set_token(os.environ.get("BUGBUG_BUGZILLA_TOKEN"))
+BUGZILLA_TOKEN = os.environ.get("BUGBUG_BUGZILLA_TOKEN")
 
 logging.basicConfig(level=logging.INFO)
 LOGGER = logging.getLogger()
 
 
-def get_model(model):
-    attribute = f"bugbug_model_{model}"
+POOL = Pool(4)
+
+
+class Database(UserDict):
+    # We might want to customize the database with cache invalidation or plug
+    # it into a real database / caching service
+    pass
+
+
+MODELS_DB = defaultdict(Database)
+
+
+def schedule_bug_classification(model_name, bug_id):
```

Comment:
Shouldn't we check if a result is already available, before scheduling a new classification?

Reply:
I'm hesitant, I would say it's the responsibility of the caller to choose if there is a result already available (and if they want to schedule it anyway). Maybe we can introduce a new higher-level function that would decides if it should schedule the classification or not.

Reply:
It works for me if we put it in the caller, or a higher level function.